File: Code/Websites/DanpheEMR/Services/NewClinical/DTOs/DischargeInformation_DTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DanpheEMR.Services.NewClinical.DTOs
{
    public class DischargeInformation_DTO
    {
        public int DischargeInformationId { get; set; }
        public int DischargeTypeId { get; set; }
        public int? SubDischargeTypeId { get; set; }
        public int? CheckdById { get; set; }
        public int DoctorInchargeId { get; set; }
        public int? ResidentDrId { get; set; }
        public int? DischargeNurseId { get; set; }
        public int PatientId { get; set; }
        public int PatientVisitId { get; set; }
        public bool IsOtPatient { get; set; }
        public string OperationType { get; set; }
        public DateTime? OperationDate { get; set; }
        public int? AnaesthetistId { get; set; }
        public List<int> SelectedConsultants { get; set; }
        public string Consultant { get; set; }


    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical-new/shared/dto/discharge-information.dto.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";

export class DischargeInformation_DTO {
  DischargeInformationId: number = 0;
  DischargeTypeId: number = 0;
  SubDischargeTypeId: number = 0;
  CheckdById: number = 0;
  DoctorInchargeId: number = 0;
  ResidentDrId: number = 0;
  DischargeNurseId: number = 0;
  IsOtPatient: boolean = false;
  OperationType: string = null;
  OperationDate: string = null;
  Consultant: string = '';
  AnaesthetistId: number = 0;

  public DischargeInfoValidator: FormGroup;

  constructor() {
    const _formBuilder = new FormBuilder();
    this.DischargeInfoValidator = _formBuilder.group({
      'DischargeTypeId': ['', Validators.compose([Validators.required])],
      'DoctorInchargeId': ['', Validators.compose([Validators.required])],
      'OperationType': [''],
      'SubDischargeTypeId': [''],
      'CheckdById': [''],
      'ResidentDrId': [''],
      'DischargeNurseId': [''],
      'AnaesthetistId': [''],
      'IsOtPatient': [false],
      'OperationDate': [''],


    });
  }

  public IsValidCheck(fieldName, validator): boolean {

    if (fieldName == undefined) {
      return this.DischargeInfoValidator.valid;
    }
    else
      return !(this.DischargeInfoValidator.hasError(validator, fieldName));
  }
}
